localflavor/cu/forms.py

```diff
@@ -0,0 +1,188 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import re
+
+from django.core.exceptions import ValidationError
+from django.core.validators import EMPTY_VALUES
+from django.forms import Field, RegexField, Select
+from django.utils import six
+from django.utils.translation import ugettext as _
+
+from .choices import PROVINCE_CHOICES, PROVINCE_NORMALIZED, REGION_CHOICES, REGION_NORMALIZED
+from .validators import CUIdentityCardNumberValidator
+
+
+class CURegionField(Field):
+    """
+    A form field for a cuban region.
+    The input is validated against a dictionary which includes names and abbreviations.
+
+    It normalizes the input to the standard abbreviation for the given region.
+
+    .. versionadded:: 1.6
+    """
+
+    default_error_messages = {
+        'invalid': _('Enter a cuban region.'),
+    }
+
+    def clean(self, value):
+        super(CURegionField, self).clean(value)
+        if value in EMPTY_VALUES:
+            return ''
+        try:
+            value = value.strip().lower()
+        except AttributeError:
```

Comment:
As far as I can see you don't need to catch an `AttributeError` here.

The same applies to `CUProvinceField`